src/lazy/binary/raw/v1_1/struct.rs

```diff
@@ -117,20 +120,167 @@ impl<'top> LazyRawStruct<'top, BinaryEncoding_1_1> for LazyRawBinaryStruct_1_1<'
     }
 }
 
+enum StructType {
+    Invalid,
+    FlexSym,
+    SymbolAddress,
+}
```

Comment:
Why not limit it to `FlexSym`/`SymbolAddress` and communicate `Invalid` via a `Result<StructType>`?

Reply:
I didn't use a result, because I'd have to somehow work a Result into the different `iter` functions, and `into_iter` which felt like a much bigger ripple than I wanted to happen. I've reworked that now, so that it only holds `FlexSym`, and `SymbolAddress`. If the constructor has an unknown opcode it will panic, since the struct iterator shouldn't be created without a valid struct.